starknet/src/utils/endian.cairo

```diff
@@ -0,0 +1,52 @@
+use integer::u128_byte_reverse;
+use sx::utils::math::pow_u128;
+use traits::TryInto;
+use option::OptionTrait;
+use array::ArrayTrait;
+
+const MASK_LOW: u128 = 0xffffffffffffffff; // 2^64 - 1
+const MASK_HIGH: u128 = 0xffffffffffffffff0000000000000000; // 2^128 - 2^64
+const SHIFT: u128 = 0x10000000000000000; // 2^64 
+
+fn uint256_into_le_words(self: u256) -> (u64, u64, u64, u64) {
+    let low_low = u128_byte_reverse(self.low & MASK_LOW) / pow_u128(2_u128, 64_u8);
+    let low_high = u128_byte_reverse(self.low & MASK_HIGH);
+    let high_low = u128_byte_reverse(self.high & MASK_LOW) / pow_u128(2_u128, 64_u8);
+    let high_high = u128_byte_reverse(self.high & MASK_HIGH);
+    (
+        low_high.try_into().unwrap(),
+        low_low.try_into().unwrap(),
+        high_high.try_into().unwrap(),
+        high_low.try_into().unwrap()
+    )
+}
+
+// U64 array is little endian
+// Using counter instead of more efficient pop_front approach for looping as we need to keep track of the final index
```

Comment:
Not sure I follow this comment.
I see the break condition is `if i == self.len() - 1` ?
I think we could still do something like:

```rust
let mut low_high = 0_u64;
let mut low_low = 0_u64;
let mut high_high = 0_u64;
let mut high_low = 0_u64;

let overflow = loop {
  match self.pop_front() {
    Option::Some(num) => {
      (low_high, low_low, high_high, high_low) = uint256_into_le_words(num);
      if self.len() == 1 {
        break (high_high);
      }
        out.append(high_high);
        out.append(high_low);
        out.append(low_high);
        out.append(low_low);
    },
    Option::None => {
      // panic here would mimic the current implementation
    }
  }
};
```

Reply:
yeah works